AlbertMineSweeper.script.js

```diff
@@ -37,10 +37,16 @@ function Grid(numRow, numCol, numMines) {
 	this.cells = [];
 	for(var i = 0; i < this.numRow; i++) {
 		this.cells[i] = [];
+		$(".base").append("<div class=\"row\" id=\"row" + i.toString() + "\"><div>");
 		for(var j = 0; j < this.numCol; j++) {
 			this.cells[i][j] = new Cell();
+			$("#row"+i.toString()).append("<div class=\"cell\" data-row=\"" + i.toString() + "\" data-col=\"" + j.toString() + "\"></div>");
 		}
-	} 
+	}
+	// Adjust Dimensions of Base
+	var edgeThickness = 2 * (parseInt($(".cell").css("border-left-width")) + parseInt($(".cell").css("margin")) + parseInt($(".cell").css("padding")));
+	$(".base").height(numRow * ($(".cell").height() + edgeThickness) + 10);
+	$(".base").width(numCol * ($(".cell").width() + edgeThickness) + 10);
```

Comment:
You won't need to do these hacks if you apply the proper CSS rules to the board and cells. Delete the above 3 lines after adding these styles.
```css
.base {
	display: inline-block; /* inline block will remove the grey space to the right - that's the difference between block and inline-block */
    padding: 5px; /* this will compensate for the removing of top/left styles */
	background-color: gray;
}
.cell {
	background-color: blue;
	height: 30px;
	width: 30px;
	position: relative;
	display: inline-block; /* <------------ Remove this because you override with display: flex below */
	margin: 2px;
	padding: 2px;
	text-align: center;
	color: blue;
	left: 5px; /* <------------ Remove this */

	display: flex; 
	align-items: center; 
	justify-content: center; 
}
.row {
	display: block;
	border: 0px;
	position: relative;
	top: 5px; /* <------------ Remove this */
	display: flex;
}
```